briefcase/macos.py

```diff
@@ -52,6 +54,12 @@ def launcher_script_location(self):
         return macos_dir
 
     def build_app(self):
+        if self.dmg:
+            print("Building DMG file...")
+            dmg_name = self.formal_name.replace(' ', '_') + '.dmg'
```

Comment:
I don't think this replace is necessary; macOS filenames don't have a problem with spaces.